### PR TITLE
remove unprintable chars

### LINKED ISSUE #720: IllegalArgumentException: Unexpected char when making directions request
_Originally posted in Navigation: https://github.com/mapbox/mapbox-navigation-android/issues/622_

While calling getRoute method on navigation instance, I am having following error:

```
onFailure: navigation.getRoute()
   java.lang.IllegalArgumentException: Unexpected char 0x2122 at 27 in User-Agent value: TODO Linux/3.18.88-Reloaded™ (aarch64)
       at okhttp3.Headers$Builder.checkNameAndValue(Headers.java:338)
       at okhttp3.Headers$Builder.add(Headers.java:288)
       at okhttp3.Request$Builder.addHeader(Request.java:177)
       at retrofit2.RequestBuilder.addHeader(RequestBuilder.java:85)
       at retrofit2.ParameterHandler$Header.apply(ParameterHandler.java:79)
       at retrofit2.ServiceMethod.toRequest(ServiceMethod.java:111)
       at retrofit2.OkHttpCall.createRawCall(OkHttpCall.java:184)
       at retrofit2.OkHttpCall.enqueue(OkHttpCall.java:91)
       at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall.enqueue(ExecutorCallAdapterFactory.java:62)
       at com.mapbox.api.directions.v5.MapboxDirections.enqueueCall(MapboxDirections.java:147)
       at com.mapbox.services.android.navigation.v5.navigation.NavigationRoute.getRoute(NavigationRoute.java:70)
```

Dependecies I am using: 

 ```
 compile ('com.mapbox.mapboxsdk:mapbox-android-sdk:5.2.1@aar'){
        transitive=true
    }
    compile 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.8.0'
    implementation ('com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.8.0') {
        transitive = true
    }
```

Android Version: 8.1

**UPDATE:**

It seems device specific. My Custom ROM's kernal name includes unicode "TM" symbol. That is causing crash!

### PR DESCRIPTION
**Pull request by osana**
closes #720

### CODE REVIEW

**Comment by osana** [comment]
@danesfeder it would be great if you could test this on the phone that had this issue

File: services-core/src/main/java/com/mapbox/core/utils/ApiCallHelper.java [hunk 1]
```diff
@@ -32,6 +32,10 @@ public static String getHeaderUserAgent(@Nullable String clientAppName) {
     if (TextUtils.isEmpty(osName) || TextUtils.isEmpty(osVersion) || TextUtils.isEmpty(osArch)) {
       return Constants.HEADER_USER_AGENT;
     } else {
+      final String onlyPrintableChars = "\\P{Print}";
```
Comment (cammace) [inline services-core/src/main/java/com/mapbox/core/utils/ApiCallHelper.java]: Is this a Regex? can we make it a constant?

**Comment by cammace** [comment]
Any chance we could add a test to ensure the issues been resolved?

**Comment by osana** [comment]
@cammace changed the test to compare string to make sure non ascii codes are removed.
Found a better pattern for ascii chars: ```[^\\p{ASCII}]```

**Pull request merged**